data-prepper-core/src/test/java/org/opensearch/dataprepper/pipeline/router/RouterTest.java

```diff
@@ -190,9 +196,45 @@ void route_with_one_DataFlowComponents_And_PipelineConnector() {
             final DataFlowComponent<PipelineConnector> dataFlowComponent = mock(DataFlowComponent.class);
             pipelineDataFlowComponents.add(dataFlowComponent);
 
-            createObjectUnderTest().route(recordsIn, pipelineDataFlowComponents, pipelineComponentRecordsConsumer);
+            createObjectUnderTest().route(recordsIn, pipelineDataFlowComponents, null, pipelineComponentRecordsConsumer);
 
             verify(dataFlowComponentRouter).route(recordsIn, dataFlowComponent, recordsToRoutes, null, pipelineComponentRecordsConsumer);
         }
+
+        @Test
```

Comment:
See my comments above about requiring the strategy.

You have added two tests which verify the behavior, but the unit tests themselves have many different scenarios.

The simplest approach is to mock the strategy at the top of the whole class and have it have the current implementation.

Put something like the following in the test setup.

```
getRecordStrategy = mock(RouterGetRecordStrategy.class);
when(getRecordStrategy.getRecord(any())).thenAnswer(a -> return a.getArguments(0));
when(getRecordStrategy.getAllRecords(anyCollection())).thenAnswer(a -> return a.getArguments(0));
```

Then you can remove these tests and use that one.

You can also add a `verify` behavior to ensure the strategy is called.